workers/store-config/src/objects/OrderIdObject.ts

```diff
@@ -1,59 +1,68 @@
+        const hashArray = Array.from(new Uint8Array(hashBuffer));
+        const hashHex = hashArray
+            .map((byte) => byte.toString(16))
+            .join('')
+            .toUpperCase();
+        return hashHex;
+    }
+
+    // NB: this will not work if there are >= 1e3 requests on the same exact ms.
+    // Atomically returns and increments the unique counter stored in DO.
+    private async sequence() {
+        let val;
+        await this.state.storage.transaction(async (txn) => {
+            val = await txn.get('counter');
+            val =
+                val === undefined || val === OrderIdObject.sequenceLength
+                    ? 0
+                    : val;
+            await txn.put('counter', val + 1);
+        });
+        return val;
+    }
+
     // Handle HTTP requests from clients.
-    async fetch(    
-        request: Request
-    ) {
+    async fetch(request: Request) {
         const HT_PREFIX = this.env.NODE_ENV === 'PROD' ? 'HT' : 'HTT';
-        
-        // Current date, YYMMDD => 230429 - This will break in year 2100 lol
-        const date = new Date()
-            .toISOString()
-            .split('T')[0]
-            .replaceAll('-', '')
-            .slice(2);
-
-        // Fetch exisiting sequence from DO
-        const storedValue = await this.state.storage?.get('value') as string;
-
-        // Make next sequence number part, 3 digits
-        // TODO what if we exceed 999, for now we will just go directly to 1000
-        let nextSeq;
-        if (
-            !storedValue ||
-            storedValue.slice(HT_PREFIX.length, HT_PREFIX.length + 6) !== date
-        ) {
-            nextSeq = getRndInteger(2, 10);
-        } else {
-            nextSeq = parseInt(storedValue.slice(HT_PREFIX.length + 6)) + 1;
-        }
-        nextSeq = nextSeq.toString().padStart(3, '0');
-
-        // Make value
-        const result = HT_PREFIX + date + nextSeq;
-
-        // Update DO and return response
-        console.debug(`Generated Order ID: ${result}, DO value: ${storedValue}.`);
-        await this.state.storage?.put('value', result);
-        return new Response(JSON.stringify({ orderId: result }), {
+
+        // Atomically generate and put the unique identifier.
+        // Padding is log10(sequenceLength). Implicit assumption sequenceLength is power of 10.
+        const nextSeq = await this.sequence();
+        const nextSeqPadded = nextSeq
+            .toString()
+            .padStart(Math.log10(OrderIdObject.sequenceLength), '0');
+
+        // Make value.
+        const result = new Date().toISOString() + nextSeqPadded;
+
+        // Obfusciate with hash, chance of collision is 16^hashLength.
+        const hashedResult =
+            HT_PREFIX +
+            (await this.hash(result)).slice(0, OrderIdObject.hashLength);
+
+        // Return response.
+        console.debug(`Generated Order ID: ${result}, DO value: ${nextSeq}.`);
```

Comment:
This log line is not printing the actual generated orderId result

Reply:
Fixed!